AsyncDisplayKit/ASMapNode.mm

```diff
@@ -160,7 +162,11 @@ - (void)takeSnapshot
   if (!_snapshotter) {
     [self setUpSnapshotter];
   }
-  [_snapshotter cancel];
+  
+  if (_snapshotter.isLoading) {
```

Comment:
@aaronschubert0 that true? I think the set options method does destroy the snapshot or, but if you could check that the logic feels correct overall for this scenario you described which is certainly an important one, that would be reassuring!

Apologies that I have not gotten to your tvOS diff yet, hope to do so soon.